pkg/storaged/btrfs/btrfs-tool.py

```diff
@@ -241,15 +231,39 @@ def cmd_poll(opt_mount):
     sys.stdout.flush()
 
 
+def unshare_mounts():
+    # The os.unshare function is available since Python 3.12, but we
+    # still need to support older Pythons.
+    if "unshare" in os.__dict__:
+        os.unshare(os.CLONE_NEWNS)
+    else:
+        libc = ctypes.CDLL(None)
+        libc.unshare.argtypes = [ctypes.c_int]
+        get_errno_loc = libc.__errno_location
+        get_errno_loc.restype = ctypes.POINTER(ctypes.c_int)
+        ret = libc.unshare(2**17)
+        if ret != 0:
+            errno = get_errno_loc()[0]
+            raise OSError(errno, os.strerror(errno))
+
+
 def cmd_do(uuid, cmd):
     debug(f"DO {uuid} {cmd}")
-    with mount_database() as db:
-        filesystems = list_filesystems()
-        for fs in filesystems.values():
-            if fs['uuid'] == uuid:
-                mp = force_mount_point(db, fs, opt_repair=True)
-                with context_chdir(mp):
-                    subprocess.check_call(cmd)
+    filesystems = list_filesystems()
+    for fs in filesystems.values():
+        if fs['uuid'] == uuid:
+            path = tempfile.mkdtemp()
+            dev = fs['devices'][0]
+            try:
+                unshare_mounts()
+                subprocess.check_call(["mount", "--make-rprivate", "/"])
+                subprocess.check_call(["mount", dev, path])
+                try:
+                    subprocess.check_call(cmd, cwd=path)
+                finally:
+                    subprocess.check_call(["umount", path])
+            finally:
+                os.rmdir(path)
```

Comment:
This would fail if the umount fails, and then cause a crash. [tempfile.TemporaryDirectory()](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) is a bit easier to use and it may also tolerate cleanup failures. Or you use a static /run/cockpit/btrfs mount point? It's not visible outside of the process anyway.

Reply:
[tempfile.TemporaryDirectory()](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) removes all files in a directory recursively, and is that too dangerous to use, IMO.

Reply:
> This would fail if the umount fails,

Yes, we could ignore EBUSY here.

I'll try to come up with a better error exit story. I think errors during cleanup should not be silenced here, but they should not be confusing, like trying to unmount when we failed to mount, or trying to remove the mount point after failing to unmount.

Reply:
See above  - it'd be better to just completely drop all this.